bin/pg_repack.c

```diff
@@ -284,6 +280,7 @@ static pgut_option options[] =
 	{ 'b', 'D', "no-kill-backend", &no_kill_backend },
 	{ 'b', 'k', "no-superuser-check", &no_superuser_check },
 	{ 'l', 'C', "exclude-extension", &exclude_extension_list },
+	{ 'u', 'm', "max-swap-tuples", &max_tuples_before_switch },
```

Comment:
There should be a sanity check that `max_tuples_before_switch < APPLY_COUNT`

Reply:
The check in the code is against the number of tuples remaining in the log. Wouldn't your statement only be true if that number was < APPLY_COUNT + max_tuples_before_switch? I don't see how this change is going to result in early termination of that loop.

Reply:
```
		/* We'll keep applying tuples from the log table in batches
		 * of APPLY_COUNT, until applying a batch of tuples
		 * (via LIMIT) results in our having applied
		 * MIN_TUPLES_BEFORE_SWITCH or fewer tuples. We don't want to
		 * get stuck repetitively applying some small number of tuples
		 * from the log table as inserts/updates/deletes may be
		 * constantly coming into the original table.
		 */
```

No, it's the number of tuples that were fetched in that batch (identical to the number that were applied), not the total number of rows remaining in the log. Doing a full count would be slower, so repack_apply wouldn't compute that

Also see lib/repack.c

```c
 * @retval				Number of performed operations.
 */
Datum
repack_apply(PG_FUNCTION_ARGS)
```